pkg/reconciler/openshift/tektonconfig/rbac.go

```diff
@@ -75,35 +83,94 @@ func (r *rbac) cleanUp(ctx context.Context) error {
 	return nil
 }
 
+// getRbacResource returns list of namespaces where rbac resources need to be created.
+func (r *rbac) getRbacResource(ctx context.Context, n corev1.Namespace, rbacNamespaces []corev1.Namespace) ([]corev1.Namespace, error) {
+	if len(r.disableRbacResourcesForOpenshift) == 0 {
+		rbacNamespaces = append(rbacNamespaces, n)
+	}
+	for _, v := range r.disableRbacResourcesForOpenshift {
+		// check for param name and if its matches to createRbacResource then disable auto creation of RBAC resources by deleting installerSet
+		if v.Name == "createRbacResource" {
+			if v.Value == "false" {
+				if err := r.deleteInstallerSet(ctx, r.tektonConfig, componentName); err != nil {
+					return nil, err
+				}
+			} else {
+				rbacNamespaces = append(rbacNamespaces, n)
+			}
+		} else {
+			rbacNamespaces = append(rbacNamespaces, n)
+		}
+		continue
+	}
+	return rbacNamespaces, nil
+}
+
 func (r *rbac) createResources(ctx context.Context) error {
 
 	logger := logging.FromContext(ctx)
 
 	// fetch the list of all namespaces which doesn't have label
-	// `openshift-pipelines.tekton.dev/namespace-ready: <release-version>`
+	// `openshift-pipelines.tekton.dev/namespace-reconcile-version: <release-version>`
 	namespaces, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("%s != %s", namespaceVersionLabel, r.version),
 	})
 	if err != nil {
 		return err
 	}
 
+	namespacesWithVersionLabel, err := r.kubeClientSet.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
```

Comment:
why do we require this call? 🤔

Reply:
For the first time when user install OSP it adds `"openshift-pipelines.tekton.dev/namespace-reconcile-version"` to all namespace except `^(openshift|kube)-` 

and later when user update TektonConfig CR for `createRbacResources` its comes to rbac.go and don't go to any namespaces because this check https://github.com/tektoncd/operator/blob/main/pkg/reconciler/openshift/tektonconfig/rbac.go#L84-L85 

so to avoid that added another call to fetch all those namespaces which have `openshift-pipelines.tekton.dev/namespace-reconcile-version` to avoid looping over the **all Namespaces**